src/lib/kdb/kdb5.c

```diff
@@ -323,10 +323,8 @@ copy_vtable(const kdb_vftabl *in, kdb_vftabl *out)
     out->refresh_config = in->refresh_config;
     out->check_allowed_to_delegate = in->check_allowed_to_delegate;
 
-    /* Copy fields for minor version 1 (major version 6). */
-    assert(KRB5_KDB_DAL_MAJOR_VERSION == 6);
     out->free_principal_e_data = NULL;
-    if (in->min_ver >= 1)
+    if (in->maj_ver > 6 || (in->maj_ver == 6 && in->min_ver >= 1))
```

Comment:
You can just remove this conditional; all loaded DB modules will have maj_ver > 6.